nomad/structs/csi_volumes.go

```diff
@@ -0,0 +1,248 @@
+package structs
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	VolumeTypeCSI = "csi"
+)
+
+type CSIVolume struct {
+	ID         string
+	Driver     string
+	Namespace  string
+	Topology   *CSITopology
+	MaxReaders int
```

Comment:
lazy q: is this a thing that we can actually get from csi?

Reply:
Yes, ListVolumes.entries is a `[]Volume`, which contains a set of accessible topologies